app-android/src/main/java/com/ustadmobile/port/android/view/ClazzMemberListFragment.kt

```diff
@@ -265,4 +276,296 @@ class ClazzMemberListFragment() : UstadListViewFragment<PersonWithClazzEnrolment
+                enabled = uiState.fieldsEnabled,
+                onClickSort = onClickSort
+            )
+        }
+
+        item {
+            ListItem(
+                text = { Text(text = stringResource(id = R.string.teachers_literal)) }
+            )
+        }
+
+        item {
+            if (uiState.addTeacherVisible){
+                AddListItem(
+                    text = stringResource(id = R.string.add_a_teacher),
+                    onClick = onClickAddNewTeacher,
+                )
+            }
+        }
+
+        items(
+            items = uiState.teacherList,
+            key = { Pair(1, it.personUid) }
+        ){ person ->
+            ListItem (
+                modifier = Modifier.clickable {
+                    onClickEntry(person)
+                },
+                text = {
+                    Text(text = "${person.firstNames} ${person.lastName}")
+                },
+                icon = {
+                    Icon(
+                        imageVector = Icons.Filled.AccountCircle,
+                        contentDescription = null
+                    )
+                }
+            )
+        }
+
+        item {
+            ListItem(
+                text = { Text(text = stringResource(id = R.string.students)) }
+            )
+        }
+
+        item {
+            if (uiState.addStudentVisible){
+                AddListItem(
+                    text = stringResource(id = R.string.add_a_student),
+                    onClick = onClickAddNewStudent,
+                )
+            }
+        }
+
+        items(
+            items = uiState.studentList,
+            key = { Pair(2, it.personUid) }
+        ){ personItem ->
+            StudentListItem(
+                person = personItem,
+                onClick = onClickEntry
+            )
+        }
+
+        item {
+            ListItem(
+                text = { Text(text = stringResource(id = R.string.pending_requests)) }
+            )
+        }
+        
+        items(
+            items = uiState.pendingStudentList,
+            key = { Pair(3, it.personUid) }
+        ){ pendingStudent ->
+            PendingStudentListItem(
+                person = pendingStudent,
+                onClick = onClickPendingRequest
+            )
+        }
+    }
+}
+
+@OptIn(ExperimentalMaterialApi::class)
+@Composable
+fun AddListItem(
+    text: String,
+    onClick: () -> Unit,
+){
+    ListItem(
+        modifier = Modifier.clickable {
+            onClick()
+        },
+        icon = {
+               Icon(
+                   Icons.Filled.PersonAdd,
+                   contentDescription = ""
+               )
+        },
+        text = { Text(text = text) }
+    )
+}
+
+ @OptIn(ExperimentalMaterialApi::class)
+ @Composable
+ fun StudentListItem(
+     person: PersonWithClazzEnrolmentDetails,
+     onClick: (PersonWithClazzEnrolmentDetails) -> Unit,
+ ){
+
+     val statusColor = if ((person.attendance/100) >=
```

Comment:
Please create a top level function called colorForAttendanceStatus(attendance) because we are using this in other screens. You can put this in a separate file in com.ustadmobile.port.android.view.util - we can call it ColorForAttendanceStatus.

Please make a similar function for React.